internal/addrs/resource.go

```diff
@@ -459,6 +459,9 @@ func (v ConfigResource) CheckableKind() CheckableKind {
 	return CheckableResource
 }
 
+func (r ConfigResource) configRemovableSigil() {
```

Comment:
What is this function supposed to do?

Reply:
We can see those empty sigil functions in different places in the codebase (for reference, the [absMovableSigil()](https://github.com/opentofu/opentofu/blob/00320dcd00c7011d1025373d63bc4c3002a306d1/internal/addrs/moveable.go#L17)). They are always empty, and from what I understand, their usage is to make sure only specific structs can implement the interface. 
For example, although on a technical level, different address types might be able to implement the `AbsMoveable` interface, logically it's not the case. Only some address types can really be treated as moveable addresses. The address type [ModuleCall](https://github.com/opentofu/opentofu/blob/00320dcd00c7011d1025373d63bc4c3002a306d1/internal/addrs/module_call.go#L14) can technically fall under the `AbsMoveable` interface, but in implementation `tofu` doesn't want to treat this address as a movable address. That's why it doesn't define the `absMoveableSigil()`.
Same logic for the `ConfigRemovable` interface.
I'm not a GO professional, so I'm not entirely sure if this is considered a best practice or not.

Reply:
I believe you are correct, let's keep it this way even if I personally feel somewhat uncomfortable with this technique. (It feels like a hack to get a `final` keyword-equivalent functionality in Go.)

Reply:
Maybe we could add a comment here to mention why it's there. So it doesn't get removed in the future.

Maybe something like `empty function to fulfil the requirements of X interface`